app/rankers/algorithms.py

```diff
@@ -60,6 +60,29 @@ def top45(items: RankableListType) -> RankableListType:
     return items[:45]
 
 
+def top1_topics(slates: List['SlateConfigModel'], personalized_topics: PersonalizedTopicList) -> List['SlateConfigModel']:
+    """
+    returns the lineup with only the top topic slate included
+    :param slates: initial list of slate configs
+    :param personalized_topics: recit response including sorted list of personalized topics
+    :return: list of slate configs with only the top topic slate
+    """
+
+    return __personalize_topic_slates(slates, personalized_topics, topic_limit=1)
+
+
+def top3_topics(slates: List['SlateConfigModel'], personalized_topics: PersonalizedTopicList) -> List[
+    'SlateConfigModel']:
+    """
+    returns the lineup with only the top 3 topic slates included
+    :param slates: initial list of slate configs
+    :param personalized_topics: recit response including sorted list of personalized topics
+    :return: list of slate configs with only the top 3 topic slate
+    """
+
+    return __personalize_topic_slates(slates, personalized_topics, topic_limit=3)
+
```

Comment:
I'm on board with this, especially since we're talking about an API surface of 2 that both point back to one implementation.

If we start having 3 or more `topN_topics` rankers, it might be worth putting our heads together on a more generalizable ranker schema.

Reply:
+1